app/screens/home/search/results/file_options/file_options.tsx

```diff
@@ -0,0 +1,210 @@
+import VideoFile from '@components/files/video_file';
+import FormattedDate from '@components/formatted_date';
+import OptionItem from '@components/option_item';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import CopyPublicLink from '@screens/gallery/footer/copy_public_link';
+import DownloadWithAction from '@screens/gallery/footer/download_with_action';
+import {getFormattedFileSize, isImage, isVideo} from '@utils/file';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+const format = 'MMM DD YYYY HH:MM A';
+
+export const TOAST_MARGIN_BOTTOM = 40;
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            marginTop: -20,
+        },
+        headerContainer: {
+            float: 'left',
+            marginTop: 20,
+            marginBottom: 8,
+        },
+        fileIconContainer: {
+            marginLeft: -10,
+            marginBottom: 8,
+            alignSelf: 'flex-start',
+        },
+        nameText: {
+            color: theme.centerChannelColor,
+            ...typography('Heading', 400, 'SemiBold'),
+        },
+        imageVideo: {
+            marginLeft: 10,
+            height: 72,
+            width: 72,
+        },
+        optionContainer: {
+            marginHorizontal: -20,
+            paddingHorizontal: 20,
+        },
+        infoContainer: {
+            marginVertical: 8,
+            alignItems: 'center',
+            flexDirection: 'row',
+        },
+        infoText: {
+            flexDirection: 'row',
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 200, 'Regular'),
+        },
+        date: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 200, 'Regular'),
+        },
+        toast: {
+            marginTop: 100,
+            alignItems: 'center',
+        },
+    };
+});
+
+type Props = {
+    canDownloadFiles: boolean;
+    enablePublicLink: boolean;
+    fileInfo: FileInfo;
+}
+const FileOptions = ({fileInfo, canDownloadFiles, enablePublicLink}: Props) => {
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+    const intl = useIntl();
+    const ref = useRef<any>();
+    const serverUrl = useServerUrl();
+    const [action, setAction] = useState<GalleryAction>('none');
+
+    const galleryItem = {...fileInfo, type: 'image'} as GalleryItemType;
+    const size = getFormattedFileSize(fileInfo.size);
+
+    const handleDownload = useCallback(() => {
+        setAction('downloading');
+    }, []);
+
+    const handleCopyLink = useCallback(() => {
+        setAction('copying');
+    }, []);
+
+    const handlePermalink = useCallback(async () => {
+        showPermalink(serverUrl, '', fileInfo.post_id, intl);
+    }, [fileInfo]);
+
+    const renderIcon = () => {
```

Comment:
to be extracted in its own file as component

Reply:
added to https://mattermost.atlassian.net/browse/MM-45617